__tests__/components/editor/ValueConstraints.test.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+import { mount } from 'enzyme';
+import ValueConstraints from '../../../src/components/editor/ValueConstraints'
+
+const jsdom = require("jsdom");
+
+setUpDomEnvironment();
+
+const vcProps = {
+  "propertyTemplates": [
+    {
+      "valueConstraint": {
+        "valueTemplateRefs": [
+          "profile:bf2:Monograph:Work"
+        ],
+        "useValuesFrom": [
+          "http://id.loc.gov/foo/works"
+        ],
+        "valueDataType": {
+          "dataTypeURI": "http://id.loc.gov/ontologies/bibframe/Instance"
+        },
+        "defaults": [
+          {
+            "defaultURI": "http://id.loc.gov/foo/instance/mono",
+            "defaultLiteral": "foo"
+          }
+        ],
+        "mandatory": "false",
+        "repeatable": "true",
+        "editable": "true"
+      }
+    }
+  ]
+}
+
+const valueConstraints = (
+  <ValueConstraints
+    valueTemplateRefs = {vcProps.propertyTemplates[0].valueConstraint.valueTemplateRefs}
+    useValuesFrom = {vcProps.propertyTemplates[0].valueConstraint.useValuesFrom}
+    valueDataType = {vcProps.propertyTemplates[0].valueConstraint.valueDataType}
+    repeatable = {vcProps.propertyTemplates[0].valueConstraint.repeatable}
+    editable = {vcProps.propertyTemplates[0].valueConstraint.editable}
+    defaults = {vcProps.propertyTemplates[0].valueConstraint.defaults} >
+  </ValueConstraints>
+)
+
+let wrapper = mount(valueConstraints)
+
+describe('<ValueConstraints />', () => {
+  it('has div with class "ValueConstraints"', () => {
+    expect(valueConstraints.props.editable).toEqual('true')
+    expect(valueConstraints.props.repeatable).toEqual('true')
+    expect(wrapper.find('h6').text()).toEqual('VALUE CONSTRAINTS')
+    expect(wrapper.find('div.ValueConstraints > ul > li')).toHaveLength(5)
+  })
+})
+
+describe('<ValueTemplateRefs />', () => {
+  it('has valueTemplateRefs', () => {
+    expect(wrapper.find('ValueTemplateRefs > div > ul > li').text()).toEqual('profile:bf2:Monograph:Work')
+  })
+})
+
+describe('<UseValuesFrom />', () => {
+  it('has valueTemplateRefs', () => {
+    expect(wrapper.find('UseValuesFrom > div > ul > li').text()).toEqual('http://id.loc.gov/foo/works')
+  })
+})
+
+describe('<ValueDataTypes />', () => {
+  it('has valueTemplateRefs', () => {
+    expect(wrapper.find('ValueDataTypes > div > ul > li').text()).toEqual('data type URI: http://id.loc.gov/ontologies/bibframe/Instance')
+  })
```

Comment:
given the logic added, I would like to see:
- a test showing the defaults are always rendered (?)
- a test showing that it does NOT render anything when a constraint is missing.
- a test showing a remark or DTU (the two properties in the ValueConstraints component itself)

Also, I'm trying to mentally map the tests to the props and/or the code and I'm confused.  Is this because it's confusing?  Would we want to clarify which props are output by which components?   Or anything along those lines?

Reply:
I have a fuller test suite now. I am clearing out the wrapper of all values to show that the child nodes are not rendered, as well as adding in tests for missing json.